modules/database/test/std/rec/printfTest.c

```diff
@@ -0,0 +1,551 @@
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_LONG, 728839938);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_s_format(void){
+
+    const char format_string[] = "Format test string %d %s";
+    const char result_string[] = "Format test string 7 Molly III";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_SHORT, 7);
+
+    /* set value on inp1 */
+    testdbPutFieldOk("test_printf_inp1_rec.VAL", DBF_STRING, "Molly III");
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 4
+}
+
+static void test_plus_flag(void){
+
+    const char format_string[] = "Format test string %+d";
+    const char result_string[] = "Format test string +7";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_SHORT, 7);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_minus_flag(void){
+
+    const char format_string[] = "Format test string %-10d";
+    const char result_string[] = "Format test string 18        ";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_SHORT, 18);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_space_flag(void){
+
+    const char format_string[] = "Format test string % d";
+    const char result_string[] = "Format test string  12";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_SHORT, 12);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_hash_flag(void){
+
+    const char format_string[] = "Format test string %#o";
+    const char result_string[] = "Format test string 014";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_SHORT, 12);
```

Comment:
Picky, but this might be a clearer test if (as above) you specify this as an octal value `014`

Reply:
Resolved in https://github.com/epics-base/epics-base/pull/340/commits/28ecf0af3c3f2585f1571215c8b16a15ccbcab37